packages/babel-helper-module-transforms/src/normalize-and-load-metadata.js

```diff
@@ -57,6 +57,38 @@ export function hasExports(metadata: ModuleMetadata) {
   return metadata.hasExports;
 }
 
+/**
+ * Check if the module has a default export (or reexport) and no other named exports (or reexports).
+ */
+export function hasDefaultExportOnly(metadata: ModuleMetadata) {
+  const { local, source } = metadata;
+
+  if (local.size === 0) {
+    const sourceReexports = new Map();
+    for (const [, srcMetadata] of source) {
```

Comment:
Could all of this be replaced with something like:

```
if (source.length > 1) return false;

const [, srcMetadata] = source[0];
if (srcMetadata.reexports.length > 1) return false;

const [exportName, importName] = reexports[0];
return exportName === "default";
```

There's maybe a balance to making it more readable, but it seems a shame to build a Map when we only care about the first item.

Reply:
Well this code is to analyze whether you have a `default reexport`. So I have to check all the imports, because you can have 10 imports and not 10 reexports. So I definitely need the loop. Then we need to check whether the import has a `reexport` and if the `reexport` is not a `default` reexport, we can return false, and if yes we can set `hasDefaultReexport` to `true`, but we need to continue the loop because its possible to have further reexports.

Code would be like this:
```js
let hasDefaultReexport = false;
for (const [, srcMetadata] of source) {
  const { reexports } = srcMetadata;
  if (reexports) {
    for (const [exportName, importName] of reexports) {
      if (exportName === 'default') {
        hasDefaultReexport = true;
      }
      else {
        return false;
      }
    }
  }
}

return hasDefaultReexport;
```

Reply:
The revised code is more readable. Can also remove unused `importName`.

Reply:
Pushed the changes!